processing/audio_analysis.py

```diff
@@ -31,6 +31,86 @@
+        jwtKey: The JWT key to use for the API.
+        conf: The configuration object.
+
+    Returns:
+        The API response.
+    """
+
+    # this used to work by default then  just stopped, so will explicitly add it
+    mimetypes.add_type("audio/mp4", ".m4a")
+
+    logger = logs.worker_logger("audio.analysis", recording["id"])
+
+    api = API(conf.api_url, conf.user, conf.password, logger)
+
+    input_extension = mimetypes.guess_extension(recording["rawMimeType"])
+
+    if not input_extension:
+        # Unsupported mimetype. If needed more mimetypes can be added above.
+        logger.error(
+            "unsupported mimetype. Not processing %s", recording["rawMimeType"]
+        )
+        api.report_done(recording, recording["rawFileKey"], recording["rawMimeType"])
+        return
+    new_metadata = {"additionalMetadata": {}}
+    with tempfile.TemporaryDirectory() as temp:
+        temp_path = Path(temp)
+        input_filename = temp_path / ("recording" + input_extension)
+        logger.debug("downloading recording to %s", input_filename)
+
+        api.download_file(jwtKey, str(input_filename))
+        track_info = api.get_track_info(recording["id"]).get("tracks")
+        track_info = [ t for t in track_info if not any(tag for tag in t["tags"] if tag["automatic"])]
+        recording["Tracks"] = track_info
+        filename = input_filename.with_suffix(".txt")
+        with filename.open("w") as f:
+            json.dump(recording, f)
+            
+        
+        analysis = analyse(input_filename, conf,analyse_tracks=True)
+        if analysis["species_identify"]:
+            species_identify = analysis.pop("species_identify")
+            for analysis_result in species_identify:
+                model_name = analysis_result.get("model", "Unnamed")
+                predictions = analysis_result["predictions"]
+                for i, prediction in enumerate(predictions):
+                    species = prediction["species"]
+                    confidences = prediction["likelihood"]
+                    del prediction["species"]
+                    raw_tag = None
+                    if len(confidences) == 0 and "raw_tag" in prediction:
+                        raw_tag = prediction["raw_tag"]
+                        species = [UNIDENTIFIED]
+                        confidences = [prediction["raw_confidence"]]
+
+                    for confidence, s in zip(confidences, species):
+                        prediction["confidence"] = confidence
+                        prediction["tag"] = s
+                        data = {"name": "Master"}
+                        if raw_tag is not None:
+                            data["raw_tag"] = raw_tag
+
+                        if i == 0:
+                            # just add master tag for first prediction
+                            api.add_track_tag(recording, analysis_result["track_id"], prediction, data)
+                        data["name"] = prediction["model"]
```

Comment:
Just double checking. Does each prediction contain model or is that only in analysis_result?

Reply:
ye, on more recent changes have started parsing the json into strctures so it's more obvious what should exist